### PR TITLE
439 story lab9 navbar repair

### LINKED ISSUE #439: Story: (Code Repair Component) Implement Navbar Repair
# Lab 9 Navbar Repair

As a **user**, I want **the navbar options to display icons**, so that **they are universally accessible**.

Associated Epic:

## Definition of Done

- [x] Acceptance criteria defined
- [x] Solution tasks specified
- [x] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [x] Design documents updated if needed
- [x] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [x] Acceptance criteria is met

## Acceptance Criteria

- [x] Given **there is a Home button**, when **the user enters 'house.svg' in the NavbarRepair page**, then **a house icon will appear next to the Home button in the nav bar**
- [x] Given **there is a Careers button**, when **the user enters 'briefcase.svg' in the NavbarRepair page**, then **a briefcase icon will appear next to the Careers button in the nav bar**
- [x] Given **there is a Education button**, when **the user enters 'grad_cap.svg' in the NavbarRepair page**, then **a graduation cap icon will appear next to the Education button in the nav bar**
- [x] Given **there is a Contact Us button**, when **the user enters 'phone.svg' in the NavbarRepair page**, then **a briefcase icon will appear next to the Contact Us button in the nav bar**

## Solution Tasks

- [x] Implement a React compliant UI for the repair code block for each interactive input
- [x] UI must resemble HMTL code as the user is updating the actual UI itself by inserting icons

### PR DESCRIPTION
**Pull request by dm9965**
- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

File: client/src/components/exercise/lab9/hooks/useRepairNav.js [hunk 1]
```diff
@@ -1,9 +1,81 @@
+import { NavBarData } from "../../../../constants/lab9/NavBarData";
```
Comment (AndreasLc1103) [inline client/src/components/exercise/lab9/hooks/useRepairNav.js:1]: Delete this file and use the `useDataService` hook in the the Nav Bar page component

File: client/src/components/exercise/lab9/pages/NavBarRepair.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import CodeLine from "../../../all-components/CodeBlock/Components/CodeLine";
+import Tab from "../../../all-components/CodeBlock/Components/Tab";
+import CommentText from "../../../all-components/CodeBlock/StyleComponents/CommentText";
+import CodeBlockInput from "../../../all-components/CodeBlock/Components/CodeBlockInput";
+import MultiTab from "../../../all-components/CodeBlock/Components/MultiTab";
+import PropTypes from "prop-types";
+import HTMLTag from "../../../all-components/CodeBlock/StyleComponents/HTMLTag";
+import HTMLText from "../../../all-components/CodeBlock/StyleComponents/HTMLText";
+import React from "react";
+
```
Comment (AndreasLc1103) [inline client/src/components/exercise/lab9/pages/NavBarRepair.jsx:11]: Document with doc strings

File: client/src/components/exercise/lab9/pages/Repairs/DateRepair.js [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
-import { GAME_STATES, REPAIR } from "../../../../../constants/lab9";
+import {
+  // eslint-disable-next-line no-unused-vars
+  REPAIR_DATE_REPAIR_HEADING,
+  GAME_STATES,
+  REPAIR,
+} from "../../../../../constants/lab9";
+
 /**
  * Date Repair is a component that is responsible for passing logic into the universal
```
Comment (AndreasLc1103) [inline client/src/components/exercise/lab9/pages/Repairs/DateRepair.js]: little thing but just get rid of the spacing between the two words and just do `DateRepair():` in the doc-string

File: client/src/components/exercise/lab9/pages/Repairs/NavRepairPage.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Nav Repair is a component that is responsible for passing logic into the universal
+ * repair page component, what this allows us to do is call the re-useable repair component
+ * with custom logic pertaining to that repair section.
+ * @param {String} user contains user id for data state and logging user input
+ * @returns
+ */
+
+const NavRepairPage = ({ user }) => {
+  const { data, functions } = useRepairNav(user);
```
Comment (AndreasLc1103) [inline client/src/components/exercise/lab9/pages/Repairs/NavRepairPage.jsx:20]: Change to the recommendation from earlier.

File: client/src/constants/lab9/NavBarData.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+export const NavBarData = {
```
Comment (AndreasLc1103) [inline client/src/constants/lab9/NavBarData.js:1]: just put a short doc string saying this is the user's default state when starting the exercise

**Review by AndreasLc1103 — COMMENTED** [review]
Looks good to me, just some minor updates to make this consistent with the rest of the feature branch. so if you can pull in `Development/Lab-9

File: client/src/constants/lab9/NavBarData.js (hunk 5 above)
Comment (dm9965) [reply client/src/constants/lab9/NavBarData.js:1]: Development/Lab-9 has been merged

**Review by AndreasLc1103 — APPROVED** [review]

**Pull request merged**